ttnn/cpp/ttnn/operations/ccl/all_gather/device/kernels/dataflow/worker_ring_gather_utils.hpp

```diff
@@ -361,30 +364,37 @@ FORCE_INLINE void read_wrapped_chunk_from_output_tensor(
     const AddrGen& s,
     const uint32_t num_pages,
     const uint32_t page_size,
-    bool& last_page_of_worker) {
+    bool& last_page_of_worker,
+    std::optional<uint32_t> local_l1_read_addr = {}) {
```

Comment:
I don't think we should be doing this. 

I'd prefer only one way to generate read addresses. Ideally we could templatize the argument by the type of the read address generator you want (would require wrapping cb_id in a strong type) so for now I think it's reasonable to just create two wrapper functions and pull out the inner core logic ons the address has been obtained. 

That being said, I'm curious why this is even needed. Is this to support the same reader/writer core?

Reply:
This is needed because datacopy always uses the buffer from the start index. so the cb ptr is initialized once and then reused.

I've pulled this out into two new read/write chunk functions so it doesn't clutter this one.